### ISSUE TITLE
QUESTION: Ubuntu shairport-sync configuration

### ISSUE THREAD

**Issue by ziv1234** [mikebrady/shairport-sync#523]
Hi,

I installed shairport-sync on my Ubuntu 17.04. I used apt-get install shairport-sync. I have an issue that I am trying to resolve with connecting to the pulseaudio system, which I wrote about in another thread as it seems another user has similar issues.

Many of the guides around shairport-sync involve building the code directly from github. I prefer to use the Ubuntu one if possible, so I can get updates. However, I am having a challenge seeing which command line options were used on the "configure" command with the standard package. 

Appreciate any help and will post the answer when I find a solution to my issue...

Thanks!

**Comment by ktb83** [comment]
shairport-sync-2.8.6/debian/rules:
```
                --with-dummy \
                --with-stdout \
                --with-pipe \
                --with-ssl=openssl \
                --with-soxr \
                --with-metadata \
                --with-avahi \
                --with-alsa \
                --with-pulseaudio \
                --with-piddir=/run/shairport-sync \
                --without-configfiles
```

**Comment by zivkedem** [comment]
Thanks!

Does the without-config-files indicate that it doesn't read the
/etc/shairport-sync.conf?

On 8 May 2017, at 18:23, KTB <notifications@github.com> wrote:

shairport-sync-2.8.6/debian/rules:

                --with-dummy \
                --with-stdout \
                --with-pipe \
                --with-ssl=openssl \
                --with-soxr \
                --with-metadata \
                --with-avahi \
                --with-alsa \
                --with-pulseaudio \
                --with-piddir=/run/shairport-sync \
                --without-configfiles

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/mikebrady/shairport-sync/issues/523#issuecomment-299898602>,
or mute the thread

**Comment by ktb83** [comment]
I don't think so. I believe that is used because the Debian package doesn't use the default behavior of the Makefile which is to copy the included shairport-sync.conf to $(DESTDIR)$(sysconfdir)/shairport-sync.conf.sample and also replace $(DESTDIR)$(sysconfdir)/shairport-sync.conf if it already exists. The Debian package just copies shairport-sync.conf to /etc/.

**Comment by mikebrady** [comment]
BTW, a new PulseAudio back end has been added in the `development` branch. It might be worth looking at.

**Comment by zivkedem** [comment]
Interesting. Do you know how soon these usually reach the stable branches?
I will certainly try it out



*From:* Mike Brady [mailto:notifications@github.com]
*Sent:* Thursday, May 18, 2017 2:15 PM
*To:* mikebrady/shairport-sync <shairport-sync@noreply.github.com>
*Cc:* Ziv Kedem <ziv@zerto.com>; Author <author@noreply.github.com>
*Subject:* Re: [mikebrady/shairport-sync] QUESTION: Ubuntu shairport-sync
configuration (#523)



BTW, a new PulseAudio back end has been added in the development branch. It
might be worth looking at.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/mikebrady/shairport-sync/issues/523#issuecomment-302375011>,
or mute the thread

**Comment by mikebrady** [comment]
A few weeks, I guess. It depends on the level of bugs detected. If there aren't many detected or reported, it might be a bit quicker.

**Comment by zivkedem** [comment]
Thanks! Installed it and it works great. I configured it without the
–with-also to be sure that it doesn’t hijack the sound card.



So far works great. Will let you know if I run into any issues.



How can I know when this is available from the Ubuntu repository? I prefer
to have my packages from there and then it gets updated on an ongoing
basis. No urgency, but this will let me tap in your continued development.



Great work BTW!!!



*From:* Mike Brady [mailto:notifications@github.com]
*Sent:* Thursday, May 18, 2017 4:48 PM
*To:* mikebrady/shairport-sync <shairport-sync@noreply.github.com>
*Cc:* Ziv Kedem <ziv@zerto.com>; Author <author@noreply.github.com>
*Subject:* Re: [mikebrady/shairport-sync] QUESTION: Ubuntu shairport-sync
configuration (#523)



A few weeks, I guess. It depends on the level of bugs detected. If there
aren't many detected or reported, it might be a bit quicker.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/mikebrady/shairport-sync/issues/523#issuecomment-302408653>,
or mute the thread

**Comment by mikebrady** [comment]
Thanks. I'm not sure how long it takes to get into the Debian or Ubuntu repositories – it's not actually me that does that. Usually, I think it's in the order of a few months after a release. [I edited you comment slightly BTW – I hope you don't mind]

**Comment by zivkedem** [comment]
Thanks. Appreciate it. Will check when I can move from my privately
compiled one to the official Ubuntu one.



One thing to note. I configured it with a different pid file directory but
it seems like it ignores that config



*From:* Mike Brady [mailto:notifications@github.com]
*Sent:* Friday, May 19, 2017 10:36 AM
*To:* mikebrady/shairport-sync <shairport-sync@noreply.github.com>
*Cc:* Ziv Kedem <ziv@zerto.com>; Author <author@noreply.github.com>
*Subject:* Re: [mikebrady/shairport-sync] QUESTION: Ubuntu shairport-sync
configuration (#523)



Thanks. I'm not sure how long it takes to get into the Debian or Ubuntu
repositories – it's not actually me that does that. Usually, I think it's
in the order of a few months after a release. [I edited you comment
slightly BTW – I hope you don't mind]

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/mikebrady/shairport-sync/issues/523#issuecomment-302631741>,
or mute the thread

**Comment by mikebrady** [comment]
Okay, thanks. I'll check that. I think it may need a little more "house training".

**Comment by mikebrady** [comment]
Hi there. I checked the `--with-piddir` configuration directive with `--with-piddir=/run/shairport-syncy` and it's working alright. It does not create the folder(s), nor does it give them appropriate ownership or permission, but it's not supposed to...

**Comment by zivkedem** [comment]
I think that this is the default. When I used as the piddir /home/ziv it
still went to look for it in /run/Shairport-sync

On 21 May 2017, at 6:35, Mike Brady <notifications@github.com> wrote:

Hi there. I checked the --with-piddir configuration directive with
--with-piddir=/run/shairport-syncy and it's working alright. It does not
create the folder(s), nor does it give them appropriate ownership or
permission, but it's not supposed to...

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/mikebrady/shairport-sync/issues/523#issuecomment-302928427>,
or mute the thread
